cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -257,6 +252,39 @@ where
     }
 }
 
+/// Construct a cache if one should be used.
+///
+/// Errors if there's an ingress URI, no signing key, and metadata_only is
+/// false.
+fn determine_cache(
+    metadata_only: bool,
+    ingress_uri: Option<Url>,
+    key_file: Option<PathBuf>,
+) -> Result<Option<NixCopyCache>, PublishError> {
+    if metadata_only {
+        return Ok(None);
+    }
+    match (ingress_uri, key_file) {
+        (Some(ingress_uri), Some(key_file)) => Ok(Some(NixCopyCache {
+            url: ingress_uri,
+            key_file,
+        })),
+        (Some(_), None) => Err(PublishError::Catchall(
```

Comment:
(non-blocking) Why not a specific variant for this new error?

Reply:
I'm not matching on the error anywhere so I don't think it's worth giving it its own type/printer in errors.rs